src/collective/fingerpointing/browser/logview.pt

```diff
@@ -21,9 +21,73 @@
     </metal:title>
 
     <metal:content-core metal:fill-slot="content-core">
-      <pre tal:condition="view/available" tal:content="view/get_audit_log" />
-
-      <p tal:condition="not:view/available" i18n:translate="">
+      <tal:block tal:condition="python:view.available"
+                 tal:define="dummy python:view.build_audit_log()">
+        <form method="get"
+              action=""
+              tal:attributes="action python:view.base_url">
+          <div class="field">
+            <input type="hidden"
+                   name="size"
+                   value=""
+                   tal:attributes="value python:view.size" />
+            <input type="text"
+                   name="query"
+                   value=""
+                   class="searchButton"
+                   placeholder="Enter a search term."
+                   i18n:attributes="placeholder"
+                   tal:attributes="value python:view.query" />
+          </div>
+          <div class="formControls button-field context">
+            <input type="submit" value="filter" i18n:attributes="filter"/>
+          </div>
+        </form>
+        <p class="fingerpointing-pagination field">
+          <!-- Needs some love CSS-wise -->
+          [<a href=""
+              class="first"
+              i18n_translate=""
+              tal:attributes="href python:view.pagination_url('first')">first</a>]
+          [<a href="#"
+              class="previous"
+              i18n_translate=""
+              tal:attributes="href python:view.pagination_url('prev')">previous</a>]
+          <span class="current-page" tal:content="python:view.page">#</span>
+          [<a href="#"
+              class="last"
+              i18n_translate=""
+              tal:attributes="href python:view.pagination_url('next')">next</a>]
```

Comment:
this must be disabled of no next page is available.

Reply:
so, I do not now if there is a next page.... that's a bit a problem if one want to avoid to read all logs into memory.

Reply:
sorry I am wrong, I confused next and last.